tools/purge_omsagent.sh

```diff
@@ -0,0 +1,407 @@
+#! /bin/bash
+
+SOLO_LAD_DIR=/var/lib/waagent/Microsoft.Azure.Diagnostics.LinuxDiagnostic-*
+OMS_LAD_DIR=/var/opt/microsoft/omsagent/LAD
+
+OMS_ONBOARD_AGENT=onboard_agent.sh
+
+WAAGENT_DIR=/var/lib/waagent
+WAAGENT_XML=$WAAGENT_DIR/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux.*.manifest.xml
+
+STOPPED_WAAGENT=0
+
+# error codes
+UNSUPPORTED_PKG_INSTALLER=66
+UNSUPPORTED_SERVICE_CONTROLLER=67
+UNSUPPORTED_WAAGENT_NAME=68
+ONBOARD_AGENT_MISSING=69
+
+
+# $1 - exit status
+call_exit()
+{
+    echo "Purge script exiting with code $1"
+    exit $1
+}
+
+ulinux_detect_installer()
+{
+    INSTALLER=""
+
+    # Detect based on distribution
+    if [ -f "/etc/debian_version" ]; then # Ubuntu, Debian
+        INSTALLER="DPKG"
+    elif [ -f "/etc/redhat-release" ]; then # RHEL, CentOS, Oracle
+        INSTALLER="RPM"
+    elif [ -f "/etc/os-release" ]; then # Possibly SLES, openSUSE
+        grep PRETTY_NAME /etc/os-release | sed 's/PRETTY_NAME=//g' | tr -d '="' | grep -qi suse
+        if [ $? == 0 ]; then
+            INSTALLER="RPM"
+        fi
+    fi
+
+    # Fall back on detection via package manager availability
+    if [ "$INSTALLER" = "" ]; then
+        if [ -x "$(command -v dpkg)" ]; then
+            INSTALLER="DPKG"
+        elif [ -x "$(command -v rpm)" ]; then
+            INSTALLER="RPM"
+        fi
+    fi
+
+    if [ "$INSTALLER" = "" ]; then
+        echo "Error: This system does not have a supported package manager" >&2
+        echo "Supported Sytems: 'DPKG' & 'RPM'" >&2
+        echo "Cannot purge without a supported package manager" >&2
+        call_exit $UNSUPPORTED_PKG_INSTALLER
+    fi
+}
+
+ulinux_detect_servicecontroller()
+{
+    CONTROLLER=""
+
+    # Detect based on systemd
+    if which systemctl > /dev/null 2>&1; then
+        CONTROLLER="systemctl"
+    
```

Comment:
nit no newline to make unified if-else block

Reply:
Fixed in this commit: https://github.com/microsoft/OMS-Agent-for-Linux/pull/1255/commits/6084ab38cd50be252158f02bd813a2b094ffb573